### ISSUE TITLE
Setter return type must be Unit & '@set:' annotations could be applied only to mutable properties

### ISSUE THREAD

**Issue by saibotma** [jOOQ/jOOQ#12148]
### Expected behavior
Generated code compiles

### Actual behavior
Compilation fails with error messages: 

Setter return type must be Unit
<img width="426" alt="Screenshot 2021-07-11 at 16 53 09" src="https://user-images.githubusercontent.com/26022902/125199870-8291c700-e268-11eb-9a7a-ee60590ddb2c.png">

'@set:' annotations could be applied only to mutable properties
<img width="418" alt="Screenshot 2021-07-11 at 17 02 57" src="https://user-images.githubusercontent.com/26022902/125200136-de108480-e269-11eb-8f76-f5c6b039607f.png">

### Steps to reproduce the problem
Using nu.studer.jooq gradle (version 6.0) plugin:
```
jooq {
    version.set(jooqVersion)

    configurations {
        create("main") {
            generateSchemaSourceOnCompilation.set(true)

            jooqConfiguration.apply {
                logging = org.jooq.meta.jaxb.Logging.WARN
                jdbc.apply {
                    driver = "org.postgresql.Driver"
                    url = postgresUrl
                    user = postgresUser
                    password = postgresPassword
                }
                generator.apply {
                    name = "org.jooq.codegen.KotlinGenerator"
                    database.apply {
                        name = "org.jooq.meta.postgres.PostgresDatabase"
                        inputSchema = "public"
                        includes = ".*"

                        forcedTypes.addAll(
                            arrayOf(
                                ForcedType()
                                    .withUserType("java.time.Instant")
                                    .withConverter("app.appella.persistence.jooq.InstantConverter")
                                    // "\\s*" stands for multiple spaces
                                    .withIncludeTypes("timestamp\\s*with\\s*time\\s*zone"),
                                ForcedType()
                                    .withUserType("app.appella.rrule.LocalRRule")
                                    .withConverter("app.appella.persistence.jooq.RRuleConverter")
                                    .withIncludeExpression("rrule"),
                                ForcedType()
                                    .withUserType("java.awt.Color")
                                    .withConverter("app.appella.persistence.jooq.ColorConverter")
                                    .withIncludeExpression(".*\\_color\$")
                            )
                        )
                    }
                    generate.apply {
                        isDeprecated = false
                        isImmutablePojos = true
                        isFluentSetters = true
                        isPojosEqualsAndHashCode = true
                    }
                    target.apply {
                        packageName = "app.appella.persistence.postgres.jooq"
                    }
                    strategy.name = "org.jooq.codegen.DefaultGeneratorStrategy"
                }
            }
        }
    }
}
```

- Create a table in database.
- Run jOOQ generator.
- Try to compile with Kotlin 1.5.0.

### Versions

- jOOQ: 3.15.0
- Java: 11
- Database (include vendor): Postgres 11.5
- OS: MacOS
- JDBC Driver (include name if inofficial driver): 42.2.23

**Comment by lukaseder** [comment]
Thanks a lot for your report. The problem is in the combination of `immutablePojos` and `fluentSetters`. We don't have any setters in immutable pojos, so the annotation must not be generated, of course.

Will fix right away.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.1 (https://github.com/jOOQ/jOOQ/issues/12153), thanks again for your report!
